packages/frontend/utils/getLocalProvider.ts

```diff
@@ -0,0 +1,17 @@
+import { Chain } from "wagmi";
+import { publicProvider } from "wagmi/providers/public";
+
+/**
+  @dev Get the localProvider for given chain
+  @param chain : specific chain for localProvider
+  @return public provider for passed chain
+**/
+export default function getLocalProvider(chain: Chain) {
+  const chainProviderFunc = publicProvider();
+
+  const chainProviderObj = chainProviderFunc(chain);
+
+  const provider = chainProviderObj?.provider();
+
```

Comment:
Personally, I'd only keep this last new line, and remove the others new lines above.

```ts
  const chainProviderFunc = publicProvider();
  const chainProviderObj = chainProviderFunc(chain);
  const provider = chainProviderObj?.provider();

  return provider;
```